packages/opentelemetry-test-utils/src/resource-assertions.ts

```diff
@@ -335,26 +335,26 @@ export const assertEmptyResource = (resource: Resource) => {
   assert.strictEqual(Object.keys(resource.attributes).length, 0);
 };
 
+/**
+ * Assert that the `resource` has at least one known attribute with the given
+ * `prefix`. By "known", we mean it is an attribute defined in semconv.
+ */
 const assertHasOneLabel = (prefix: string, resource: Resource): void => {
```

Comment:
Reviewer note: I changed this function usage to take a `prefix` in lowercase, i.e. what the actual semconv string values are.

I also changed the implementation to something that reads clearer for me.
FWIW, the `.reduce(...)` usage in the removed code had a small bug in it: it didn't provide an initial accumulator value, so the assert message wasn't correct.

Reply:
🤔  I'd prefer to do this in a way that doesn't require bring in semconv as * - isn't service name an expected "known" attribute here that could be safely used in the test? This isn't blocking since it's a test-utils package though.

Reply:
@JamieDanielson I'm not sure exactly what you mean. Take the call `assertHasOneLabel('k8s', resource);`, for example. This function needs to see if `resource.attributes` has at least one attribute that starts with `k8s.` and is an attribute value exported by `@opentelemetry/semantic-conventions`. How else could that be done without importing semconv as `*`?

Reply:
Hmm looking at this again I'm not sure what I had in mind 😅  I was thinking that we could scope the imports of attributes to only pull in a few for each test, but it seems like it could unwieldy. Probably best to ignore me on this one.